### ISSUE TITLE
any advice on how to convert this to a regression model?

### ISSUE THREAD

**Issue by tybrad11** [Kamnitsask/deepmedic#98]
For example, predicting pixelwise CT numbers instead of tissue classes. I suspect I would have to change the final classification layer to a linear model in line 936 of cnn3d.py and change the loss function to mean squared error. Anything else? As I don't see a linear layer class in layers.py, I suspect I'd have to create that.

**Comment by Kamnitsask** [comment]
Hello,

I believe there will be a couple more things to do, but they should not be too difficult.

- implement the final regresssion layer instead of the current softmax, with a corresponding loss.

- adjust the cnn3d.py-> compile(Train/Validation/Test)Function. For example, casting the y labels to int is no longer needed in both functions (search for intCastSharedLabelsYTrain = T.cast(self.sharedLabelsYTrain, 'int32') and same for val). Not sure what other small change may be needed.

- Adjust the metrics that are reported and logged. These are computed in theano in the above compile(train/val/test) functions, eg by calling softmax's getRpRnTpTnForTrain0OrVal1(). Values are passed back to the training routine at logged, at deepmedic.routines.training-> search where cnnTrainModel or cnnValidateModel are called and adjust (or remove?).

- Change the sampling for training. By default, we use the labels to focus half the sampling on foreground (labels > 0). You ll probably have to switch it to uniform. This is done in the trainConfig.cfg. There, change the following:
useDefaultTrainingSamplingFromGtAndRoi = False
typeOfSamplingForTraining = 1(uniform)

Hope the above help,
Kostas
